feat: add more gsoc 2024 projects (#7302)
jenkins-infra/jenkins.io: content/projects/gsoc/2024/projects/enhancing-an-existing-llm-model-with-domain-specific-jenkins-knowledge.adoc

---
layout: gsocproject2
title: "Enhancing an Existing LLM Model with Domain-specific Jenkins Knowledge"
goal: "To develop an app using an existing open-source LLM model with data collected for domain-specific Jenkins knowledge one can fine-tune locally and set up with a proper UI for the user to interact with"
category: AI/ML
year: 2024
status: "Selected"
sig: infra
skills:
- Python
- React.js
- LLM
- AI/ML
- Jenkins
- Ollama
- LangChain
- UI
mentors:
- "krisstern"
- "harsh-ps-2003"
- "shivaylamba"
- "gounthar"
- "laulopezreal"
student: "nouralmulhem"
links:
  idea: "/projects/gsoc/2024/project-ideas/enhancing-an-existing-llm-model-with-domain-specific-jenkins-knowledge"
  draft: "https://summerofcode.withgoogle.com/media/user/f181d5c472e4/proposal/gAAAAABmQ2Jqk1mDPvYeNs8SJRiYParUMC22oAU3gWITa5Lk-pwBLLE9Bxqzctsv110C6WNbdgLAoZ7eq-qktKTciVgFyaXR1Mz7UzybNY3ldXHEf9r6Kzk=.pdf"
  meetings: "/projects/gsoc/2024/projects/enhancing-an-existing-llm-model-with-domain-specific-jenkins-knowledge#office-hours"
---

=== Abstract

> This full-stack project focuses on a proof-of-concept (PoC) idea to fine-tune an existing open-source LLM model (such as Llama 2) with domain-specific Jenkins data to be compiled, wrangled, and processed by the contributor as a part of an AI-driven application, to develop a minimalistic UI for the user to interact with the LLM as a complete end-to-end product.

Reviewer: ```suggestion
This full-stack project focuses on a proof-of-concept (PoC) idea to fine-tune an existing open-source large language model (LLM), such as Llama 2, with domain-specific Jenkins data.
The data will be compiled, wrangled, and processed by the contributor as part of an AI-driven application.
The goal is to develop a minimalistic user interface (UI) for users to interact with the fine-tuned LLM as a complete end-to-end product.
```
Suggestion to adjust formatting to single line.

This product is to be installed and run locally on the user's laptop, with tools such as link:https://ollama.com/[Ollama] for setting up and running LLMs locally, and link:https://www.langchain.com/[LangChain] to be used as a framework to construct the LLM-powered app.

> The contributor will get to be involved in every step of the application development process, from data collection, wrangling, and processing to fine-tuning the model and developing the UI.

Reviewer: ```suggestion
The contributor will be involved in every step of the application development process, from data collection, wrangling, and processing, to fine-tuning the model and developing the user interface (UI).
```
Small suggestion in addition to @gounthar 's to chance "will get to be" to "will be"

They may also get exposed to how to package software to be distributed as a standalone application to be consumed by the end user.

=== Rationale

> Currently we do not have any AI-driven assistive technology to help Jenkins users with domain-specific knowledge. This project aims to fill that gap by developing an AI-driven application that can assist Jenkins users with such knowledge that is normally possessed by a Jenkins expert.

Reviewer 2: ```suggestion
Currently, we do not have any AI-driven assistive technology to help Jenkins users with domain-specific knowledge.
This project aims to fill that gap by developing an AI-driven application that can assist Jenkins users with the knowledge that is normally possessed by a Jenkins expert.
```

=== Implementation

> Due to the cutting-edge nature of the project, we will be combining both research and experimentation to arrive at an LLM-driven application using more than one approach to fine-tune the model with domain-specific Jenkins data. We will be using both Llama 2 and Llama 3 in various stages of the project, with the former to establish some benchmarks and the latter to experiment with the latest advancements in LLMs.

Reviewer 2: ```suggestion
Due to the cutting-edge nature of the project, we will be combining both research and experimentation to arrive at an LLM-driven application using more than one approach to fine-tune the model with domain-specific Jenkins data.
We will be using both Llama 2 and Llama 3 at various stages of the project, with the former to establish some benchmarks and the latter to experiment with the latest advancements in LLMs.
```
